src/pages/bus/shuttle/[routeId].tsx

```diff
@@ -1,20 +1,33 @@
 import { useState } from 'react';
+import React from 'react';
 import { useRouter } from 'next/router';
 import { cn } from '@bcsdlab/utils';
 import BusIcon from 'assets/svg/Bus/bus-icon-32x32.svg';
 import InformationIcon from 'assets/svg/Bus/info-gray.svg';
+import BusCoursePage from 'components/Bus/BusCoursePage';
+import { ShuttleCategoryTabs } from 'components/Bus/BusCoursePage/components/ShuttleCategoryTabs';
 import useShuttleTimetableDetail from 'components/Bus/BusCoursePage/hooks/useShuttleTimetableDetail';
 import useLogger from 'utils/hooks/analytics/useLogger';
 import useMediaQuery from 'utils/hooks/layout/useMediaQuery';
-import styles from './BusTimetableDetail.module.scss';
+import styles from './ShuttleDetailPage.module.scss';
 
-export default function BusTimetableDetail() {
+function asString(value: string | string[] | undefined): string | undefined {
+  return Array.isArray(value) ? value[0] : value;
+}
+
+export default function ShuttleDetailPage() {
   const router = useRouter();
   const { routeId } = router.query;
+
   const { shuttleTimetableDetail } = useShuttleTimetableDetail(
     routeId ? (Array.isArray(routeId) ? routeId[0] : routeId) : null,
   );
+
   const [selectedDetail, setSelectedDetail] = useState<string | null>(null);
+
+  const categoryFromURL = router.query.category;
+  const category = asString(categoryFromURL) ?? '전체';
```

Comment:
```suggestion
  // undefined인 경우는 "전체" 아니면 astring(categoryFromURL)
  const category = categoryFormURL ? asString(categoryFromURL) : '전체';
```
이런식으로하면 asString에 `undefined` 타입을 없앨 수 있을 거 같네요

Reply:
반영하였습니다 !